pkg/operator/ceph/controller/network.go

```diff
@@ -0,0 +1,429 @@
+	}
+	if retcode != 0 {
+		return ranges, errors.Errorf("ceph %q network canary ceph CSI version job returned failure code %d: stdout: %q: stderr: %q", cephNetwork, retcode, stdout, stderr)
+	}
+
+	splitOutput := strings.Split(stdout, separator())
+	if len(splitOutput) != 2 {
+		// don't log the raw outputs in case there could be sensitive info
+		return ranges, errors.Errorf("ceph %q network canary job did not return two separate outputs (instead %d)", cephNetwork, len(splitOutput))
+	}
+
+	rawNetStatus := strings.TrimSpace(splitOutput[0])
+	if rawNetStatus == "" {
+		return ranges, errors.Errorf("ceph %q network canary job did not return network status", cephNetwork)
+	}
+
+	rawIpAddrs := strings.TrimSpace(splitOutput[1])
+	if rawIpAddrs == "" {
+		return ranges, errors.Errorf("ceph %q network canary job did not return ip address info", cephNetwork)
+	}
+
+	netStatuses, err := k8sutil.ParseNetworkStatusAnnotation(rawNetStatus)
+	if err != nil {
+		return ranges, errors.Wrapf(err, "failed to parse ceph %q network canary status annotation", cephNetwork)
+	}
+	netStatus, ok := k8sutil.FindNetworkStatusByInterface(netStatuses, string(cephNetwork))
+	if !ok {
+		return ranges, errors.Errorf("failed to find ceph %q network status in canary status annotation", cephNetwork)
+	}
+
+	ipAddrs, err := k8sutil.ParseLinuxIpAddrOutput(rawIpAddrs)
+	if err != nil {
+		return ranges, errors.Wrapf(err, "failed to get IP address info for ceph %q network in canary pod", cephNetwork)
+	}
+	if len(ipAddrs) != 1 {
+		return []string{}, errors.Errorf("should have only one (found %d) 'ip address' result from network %q canary pod", len(ipAddrs), cephNetwork)
+	}
+	ifaceInfo := ipAddrs[0]
+
+	ranges, err = crossReferenceNetworkStatusAndIpResult(netStatus, ifaceInfo)
+	if err != nil {
+		return []string{}, errors.Wrapf(err, "failed to cross reference network %q canary pod status annotation and internal ip results", cephNetwork)
+	}
+
+	return ranges, nil
+}
+
+func separator() string {
+	return "===== " + nadv1.NetworkStatusAnnot + " above ===== ip address below ====="
+}
+
+// use downward api to make the network status available at /tmp/network-status
+// NOTE: env var might be preferable, but in testing, putting annotation into env var via downward
+// api had a race condition where the value was almost always empty when the container first ran,
+// even after waiting for the annotation in an init container; so use volume instead
+func networkStatusVolumeAndMount() (corev1.Volume, corev1.VolumeMount) {
+	vol := corev1.Volume{
+		Name: "network-status",
+		VolumeSource: corev1.VolumeSource{
+			DownwardAPI: &corev1.DownwardAPIVolumeSource{
+				Items: []v1.DownwardAPIVolumeFile{{
+					Path: "network-status",
+					FieldRef: &v1.ObjectFieldSelector{
+						FieldPath: `metadata.annotations['` + nadv1.NetworkStatusAnnot + `']`,
+					}}}}}}
+	mnt := corev1.VolumeMount{
+		Name:      "network-status",
+		MountPath: "/tmp",
```

Comment:
nit: What about a path like this? Maybe I just think of `/tmp` as a scratch folder for someone connecting to the pod.
```suggestion
		MountPath: "/var/lib/rook/multus",
```